graphql_compiler/schema_transformation/split_query.py

```diff
@@ -34,6 +39,340 @@ def __init__(self, query_ast):
+
+    - If the input AST starts with a cross schema vertex field, the output will be a Field object
+      representing the property field used in the stitch
+      - If a property field of the expected name already exists in the selections one level
+        above (in parent_selections), the new Field will contain any existing directives of
+        this field
+    - Otherwise, the process will be repeated on child selections (if any) of the input AST
+      - If no child selection is modified, the exact same object as the input AST will be returned
+      - If some child selection is modified, a copy of the AST with the new selections will be
+        returned
+        - If a modified child selection is a property field
+          - If a property field of the same name already exists, the new field will replace the
+            existing field
+          - Otherwise, the property field will be inserted into selections after the last existing
+            property field
+        - Otherwise, the modified selection will be appended to selections
+
+    Args:
+        query_node: SubQueryNode, whose list of child query connections may be modified to include
+                    children
+        ast: type Node, the AST that we are trying to split into child components. It is not
+             modified by this function
+        parent_selections: List[Node], containing all property fields (and possibly some vertex
+                           fields) in the level of selections that contains the input ast. It
+                           is not modified by this function
+        type_info: TypeInfo, used to get information about the types of fields while traversing
+                   the query ast
+        edge_to_stitch_fields: Dict[Tuple(str, str), Tuple(str, str)], mapping
+                               (type name, vertex field name) to
+                               (source field name, sink field name) used in the @stitch directive
+                               for each cross schema edge
+        intermediate_out_name_assigner: IntermediateOutNameAssigner, object used to generate
+                                        and keep track of names of newly created @output
+                                        directives
+
+    Returns:
+        Node object to replace the input AST in the selections one level above. If the output
+        is unchanged from the input AST, the exact same object will be returned
+    """
+    # Check if there is a split here. If so, split AST, make child query node, return property
+    # field. If not, recurse on all child selections (if any)
+    if isinstance(ast, Field):
+        parent_type_name = type_info.get_parent_type().name
+        edge_field_name = ast.name.value
+        if (parent_type_name, edge_field_name) in edge_to_stitch_fields:
+            # parent_field_name and child_field_name are names of property fields in the stitch
+            parent_field_name, child_field_name = \
```

Comment:
nit: avoid backslashes, per the Google Python style guide that we broadly follow: http://google.github.io/styleguide/pyguide.html#32-line-length

> Make use of Python’s implicit line joining inside parentheses, brackets and braces. If necessary, you can add an extra pair of parentheses around an expression.